### PR TITLE
Add Kaiming uniform init for LoRA adapters

### PR DESCRIPTION
**Pull request by michal2409**
# What does this PR do ?

Add Kaiming uniform initialization for LoRA adapters

**Collection**: nlp

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by michal2409** [comment]
jenkins

File: nemo/collections/nlp/modules/common/megatron/utils.py [hunk 1]
```diff
@@ -127,6 +127,13 @@ def init_(tensor):
     return init_
 
 
+def init_method_kaiming_uniform():
+    def init_(tensor):
+        return torch.nn.init.kaiming_uniform_(tensor, a=math.sqrt(5))
```
Comment (ShriyaPalsamudram) [inline nemo/collections/nlp/modules/common/megatron/utils.py:132]: Can the value of `a` be a parameter instead of hard coding to `math.sqrt(5)`?

Comment (michal2409) [reply nemo/collections/nlp/modules/common/megatron/utils.py]: fixed

**Comment by michal2409** [comment]
jenkins

**Comment by michal2409** [comment]
jenkins

**Review by ShriyaPalsamudram — APPROVED** [review]

**Comment by michal2409** [comment]
jenkins

**Pull request merged**